File: OneToOneCallSample/app/src/test/java/com/stringee/apptoappcallsample/stringee/receiver/CallActionOwnershipTest.java
package com.stringee.apptoappcallsample.stringee.receiver;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

public class CallActionOwnershipTest {
    @Test
    public void provisionalPushRejectTargetsSdkCallOnceItIsAvailable() {
        assertTrue(CallActionOwnership.canTargetActiveSession(
                true, true, -1, 12));
    }

    @Test
    public void sdkNotificationMustMatchTheCurrentSessionGeneration() {
        assertTrue(CallActionOwnership.canTargetActiveSession(
                true, true, 12, 12));
        assertFalse(CallActionOwnership.canTargetActiveSession(
                true, true, 11, 12));
    }

    @Test
    public void actionCannotTargetSessionWithoutPushOwnershipOrActiveCall() {
        assertFalse(CallActionOwnership.canTargetActiveSession(
                false, true, -1, 12));
        assertFalse(CallActionOwnership.canTargetActiveSession(
                true, false, -1, 12));
    }
}

File: Kotlin-OneToOneCallSample/app/src/test/java/com/stringee/kotlin_onetoonecallsample/stringee/receiver/CallActionOwnershipTest.kt
package com.stringee.kotlin_onetoonecallsample.stringee.receiver

import com.stringee.kotlin_onetoonecallsample.stringee.receiver.CallActionOwnership.canTargetActiveSession
import org.junit.Assert
import org.junit.Test

class CallActionOwnershipTest {
    @Test
    fun provisionalPushRejectTargetsSdkCallOnceItIsAvailable() {
        Assert.assertTrue(
            canTargetActiveSession(
                true, true, -1, 12
            )
        )
    }

    @Test
    fun sdkNotificationMustMatchTheCurrentSessionGeneration() {
        Assert.assertTrue(
            canTargetActiveSession(
                true, true, 12, 12
            )
        )
        Assert.assertFalse(
            canTargetActiveSession(
                true, true, 11, 12
            )
        )
    }

    @Test
    fun actionCannotTargetSessionWithoutPushOwnershipOrActiveCall() {
        Assert.assertFalse(
            canTargetActiveSession(
                false, true, -1, 12
            )
        )
        Assert.assertFalse(
            canTargetActiveSession(
                true, false, -1, 12
            )
        )
    }
}
